pallets/grants/src/lib.rs

```diff
@@ -288,34 +267,26 @@ pub mod pallet {
 	#[pallet::genesis_build]
 	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
 		fn build(&self) {
-			let grants = self
-				.vesting
-				.iter()
-				.map(|(ref who, schedules)| {
-					(
-						who.clone(),
-						schedules
-							.iter()
-							.map(|&(start, period, period_count, per_period)| VestingSchedule {
-								start,
-								period,
-								period_count,
-								per_period,
-							})
-							.collect::<Vec<_>>(),
-					)
-				})
-				.collect::<Vec<_>>();
-
-			// Create the required coins at genesis and add to storage
-			grants.iter().for_each(|(ref who, schedules)| {
-				let total_grants = schedules.iter().fold(Zero::zero(), |acc: BalanceOf<T>, s| {
+			self.vesting.iter().for_each(|(ref who, schedules)| {
+				let vesting_schedule: BoundedVec<VestingScheduleOf<T>, T::MaxSchedule> = schedules
+					.iter()
+					.map(|&(start, period, period_count, per_period)| VestingSchedule {
+						start,
+						period,
+						period_count,
+						per_period,
+					})
+					.collect::<Vec<_>>()
+					.try_into()
```

Comment:
How about going for a bounded vec at line 255 where `GenesisConfig` is defined to avoid converting from Vec to BoundedVec here?

Reply:
Looks like we can't do it ...

* Due to `trait bound constratints on serde`. And all the pallets in substrate are using `Vec<>` inside `pallet::genesis_config`, and there is no `derive` appart from `#[cfg_attr(feature = "std", derive(Default))]`

* Secondly the `type` hold inside `Vec` & `BoundedVec` containers are different. type transformation is required.

![](https://i.imgur.com/VCo62WW.png)

![](https://i.imgur.com/ECfotKk.png)